docs/brokerpak-specification.md

```diff
@@ -0,0 +1,334 @@
+# Brokerpak V1 specification
+
+This document will explain how a brokerpak is structured, and the schema it follows.
+
+A brokerpak is comprised of a versioned Terraform binary and providers for one
+or more platform, a manifest, one or more service definitions, and source code.
+Here are the contents of an example brokerpak:
+
+```
+MODE        SIZE      NAME
+drwx------  0         
+drwxr-xr-x  0         bin
+drwxr-xr-x  0         bin/linux
+drwxr-xr-x  0         bin/linux/386
+-rwxrwxr-x  75669952  bin/linux/386/terraform
+-rwxr-xr-x  35176256  bin/linux/386/terraform-provider-google_v1.19.0_x4
+drwxr-xr-x  0         bin/linux/amd64
+-rwxrwxr-x  89397536  bin/linux/amd64/terraform
+-rwxr-xr-x  41450144  bin/linux/amd64/terraform-provider-google_v1.19.0_x4
+drwx------  0         definitions
+-rw-------  5810      definitions/service-0.yml
+-rw-------  467       manifest.yml
+drwxr-xr-x  0         src
+-rw-r--r--  8082786   src/terraform-provider-google.zip
+-rw-r--r--  14418716  src/terraform.zip
+```
+
+You can create, inspect, validate, document and test brokerpaks using the `pak` sub-command.
+Run the command `gcp-service-broker pak help` for more information about creating a pak.
+
+## Manifest
+
+Each brokerpak has a manifest file named `manifest.yml`. This file determines
+which architectures your brokerpak will work on, which plugins it will use,
+and which services it will provide.
+
+### Structure
+
+**manifest.yml**
+
+| Field | Type | Description |
+| --- | --- | --- |
+| packversion* | int | The version of the manifest. MUST be `1`. Breaking changes to syntax or semantics will get an incremented packversion number. Future numbers MUST be increasing but might not be sequential. |
```

Comment:
I'm a little confused about the "MUST be `1`" part. Does mean the version must start at `1`? Seems like this could be worded a bit more clearly.

Reply:
I noticed this down below as well. It this is a "schema" for brokerpaks, then consider making that a bit more explicit in the wording.